e2e-tests/production-runtime/cypress/integration/assets.js

```diff
@@ -0,0 +1,36 @@
+describe(`webpack assets`, () => {
+  beforeEach(() => {
+    cy.intercept("/gatsby-astronaut.png").as("static-folder-image")
+    // Should load two files: normal and italic
+    cy.intercept("/static/merriweather-latin-300**.woff2").as("font")
+    cy.intercept("/static/gatsby-astronaut-**.png").as("img-import")
+    cy.visit(`/assets/`).waitForRouteChange()
+  })
+
+  // Service worker is handling requests so this one is cached by previous runs
+  if (!Cypress.env(`TEST_PLUGIN_OFFLINE`)) {
+    it(`should only create one font file (no duplicates with different hashes)`, () => {
```

Comment:
This is a bit cumbersome, but I initially tried to do what is mentioned in https://stackoverflow.com/questions/47354360/is-there-a-way-to-assert-that-a-route-has-not-been-called-in-cypress but it didn't really work.

So I've added the "italic" version to the page, too. So if we'd have duplicate entries it would first match the duplicate and not the italic version.

Reply:
In tests for static query runtime errors (not yet merged) I use 
```
cy.get("@appDataFetch.all").should("have.length", 2)
```

which I did find in https://github.com/cypress-io/cypress/issues/477#issuecomment-412617742  .

Might work for `0` case too?

Reply:
Unfortunately the `wait()` just times out, you can't "catch" it and handle gracefully. The thing you mentioned is also in the SO thread. So I could use yours in addition to mine to verify there are only two requests, but I only wait for two anyways 🤔